src/plugin/file/kml/ui/kmlnode.js

```diff
@@ -400,6 +408,16 @@ class KMLNode extends SlickTreeNode {
    */
   setImage(image) {
     this.image_ = image;
```

Comment:
Layer nodes attach listeners to the layer, and are themselves event targets. We'll want to make sure any existing node gets disposed, or these listeners will prevent the object from being garbage collected and it will be leaked.

Prior to setting the image layer reference, check if `this.layerNode` is defined and if so dispose it. You can also safely do this without checking via:
```
goog.dispose(this.layerNode);
this.layerNode = null;
```
This will also ensure the node is cleaned up in `KMLNode#disposeInternal` when the image is set to null.